qiskit_ibm/runtime/ibm_runtime_service.py

```diff
@@ -231,21 +234,26 @@ def run(
             inputs.validate()
             inputs = vars(inputs)
 
+        if image and ':' not in image:
+            raise IBMQInputValueError('"image" needs to be in form of image_name:tag')
```

Comment:
nit - the validation can be a bit more rigorous, e.g. `foo:bar:` would pass with this check